pennylane_lightning/lightning_qubit.py

```diff
@@ -131,6 +145,62 @@ def apply_lightning(self, state, operations):
 
         return np.reshape(state_vector, state.shape)
 
+    def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
+        if not CPP_BINARY_AVAILABLE:
+            return super().adjoint_jacobian(tape, starting_state, use_device_state)
+
+        if self.shots is not None:
+            warn(
+                "Requested adjoint differentiation to be computed with finite shots."
+                " The derivative is always exact when using the adjoint differentiation method.",
+                UserWarning,
+            )
+
+        if len(tape.trainable_params) == 0:
+            return np.array(0)
+
+        for m in tape.measurements:
+            if m.return_type is not Expectation:
+                raise QuantumFunctionError(
+                    "Adjoint differentiation method does not support"
+                    f" measurement {m.return_type.value}"
+                )
```

Comment:
A quick flyby comment to ask - since probabilities are essentially just a list of expectation values of projectors, would it also be fairly trivial to support differentiating probabilities? This is a very common use-case

Reply:
That should be a straight-forward extension. Once we have projectors (we can easily support this with some python-layer magic right now), the probabilities should fall out. Though , I think we can add this in a follow-up PR.

Reply:
Added as a future story

Reply:
> Once we have projectors (we can easily support this with some python-layer magic right now), the probabilities should fall out. 

Yep, I think it should be relatively straightforward! Or potentially you might not even need to support projectors, maybe a tweak to the adjoint logic in a future story/PR should allow probability gradients to fall out?